src/actions/FioActions.tsx

```diff
@@ -61,9 +62,8 @@ export function checkFioObtData(wallet: EdgeCurrencyWallet, transactions: EdgeTr
     const fioWallets = state.ui.wallets.fioWallets
     if (account == null || account.currencyConfig == null || fioWallets.length === 0) {
       checkFioObtDataCounter++
-      return setTimeout(() => {
-        dispatch(checkFioObtData(wallet, transactions))
-      }, 400)
+      await snooze(400)
+      await dispatch(checkFioObtData(wallet, transactions))
```

Comment:
This will will be an infinite recursion. In the old version, they had a `return` statement to break the loop, but that's missing here.

Instead, try:

```ts
let account: EdgeAccount
let FioWallets: string[]

// Loop until the account and wallets appear (the app may not have loaded them yet):
let loopCount = 0
while (true) {
  account = state.core.account
  fioWallets = state.ui.wallets.fioWallets
  if (account != null && account.currencyConfig != null && fioWallets.length >0) break
  if (loopCount++ > MAX_OBT_DATA_CHECKS) return
  await snooze(400)
}
```